modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy

```diff
@@ -526,34 +584,6 @@ class PodSpecBuilder {
 
     }
 
-    @PackageScope
-    Map addCpuResources(Integer cpus, Map res) {
-        if( res == null )
-            res = [:]
-
-        final req = res.requests as Map ?: new LinkedHashMap<>(10)
-        req.cpu = cpus
-        res.requests = req
-
-        return res
-    }
-
-    @PackageScope
-    Map addMemoryResources(String memory, Map res) {
-        if( res == null )
-            res = new LinkedHashMap(10)
-
-        final req = res.requests as Map ?: new LinkedHashMap(10)
-        req.memory = memory
-        res.requests = req
-
-        final lim = res.limits as Map ?: new LinkedHashMap(10)
-        lim.memory = memory
-        res.limits = lim
-
-        return res
-    }
```

Comment:
Was not this introduced by #3338? why it's gone?

Reply:
The only reason I added these methods was because I couldn't figure out how to make the code static so I needed `@CompileDynamic`, but since you figured out how to make it static, these methods aren't needed. I would have needed to add another method for disk resource. Now the code is much more concise.

Reply:
Now that I know how to use maps in a static way, I think we should try to remove all occurrences of CompileDynamic throughout the code in a separate PR.

Reply:
Still, it's important to keep changes consistent with what the PR declares.